WorkloadManagementSystem/Agent/JobAgent.py

```diff
@@ -304,8 +304,9 @@ def execute(self):
     mpTag = 'MultiProcessor' in params.get('Tags', [])
 
     if self.extraOptions:
-      params['Arguments'] += ' ' + self.extraOptions
       params['ExtraOptions'] = self.extraOptions
+      if 'Arguments' in params:
+        params['Arguments'] += ' ' + self.extraOptions
```

Comment:
Would you not want the extraOptions if there are no arguments?
```suggestion
        params['Arguments'] = (params['Arguments'] + ' ' + self.extraOptions).strip()
```
(without the `if 'Arguments'...`)

Reply:
Did you mean :

    params['Arguments'] = (params.get('Arguments', '') + ' ' + self.extraOptions).strip()

Reply:
Of course! You read my mind :)